File: _AI_Arsivi/test_41_20.py
import json
import re

with open('PROMPT_BUILDER_v8_0.html', 'r', encoding='utf-8') as f:
    content = f.read()

start = content.find('let db = {') + 9
end = content.find(';\n', start)
db_str = content[start:end]
db_str = re.sub(r'([a-zA-Z0-9_]+):', r'"\1":', db_str)
# Handle arrays with unquoted keys if necessary
db_str = db_str.replace('\'', '"')
db_str = db_str.replace('""', '"')

try:
    db = json.loads(db_str)
    entries = db['entries']
    
    # Check 41
    last_20 = [e.get('nums', []) for e in entries[:20]]
    in_last_20_41 = any(41 in d for d in last_20)
    print(f"Is 41 in last 20 draws? {in_last_20_41}")
    for i, d in enumerate(last_20):
        if 41 in d:
            print(f"41 found at index {i}")

    # Check 20 streaks
    streak_events = 0
    current_streak = 0
    draws = [e.get('nums', []) for e in reversed(entries)]
    for d in draws:
        if 20 in d:
            current_streak += 1
        else:
            if current_streak >= 2:
                streak_events += 1
            current_streak = 0
    if current_streak >= 2:
        streak_events += 1
    print(f"Streak events for 20: {streak_events}")

except Exception as e:
    print(f"JSON Error: {e}")

File: _AI_Arsivi/test_41_20.js
const fs = require('fs');
const content = fs.readFileSync('PROMPT_BUILDER_v8_0.html', 'utf-8');
const start = content.indexOf('let db = {') + 9;
const end = content.indexOf(';\n', start);
const db_str = content.slice(start, end);

// evaluate as JS object
const db = new Function('return ' + db_str)();

const df = db.entries;

const last_20 = df.slice(0, 20).map(e => e.nums);
const in_last_20_41 = last_20.some(d => d.includes(41));
console.log("Is 41 in last 20 draws?", in_last_20_41);
last_20.forEach((d, i) => {
    if (d.includes(41)) console.log(`41 found at index ${i}`);
});

let streak_events_20 = 0;
let current_streak_20 = 0;
const reversed_draws = [...df].reverse().map(e => e.nums);
for (let d of reversed_draws) {
    if (d.includes(20)) {
        current_streak_20++;
    } else {
        if (current_streak_20 >= 2) streak_events_20++;
        current_streak_20 = 0;
    }
}
if (current_streak_20 >= 2) streak_events_20++;
console.log("Streak events for 20:", streak_events_20);

// Also check K2 points for 41
// Son 10 Taban Puanı (k2'ye ekle)
const son_10_donem = df.slice(0, 10).map(e => e.nums);
let in_son_10_41 = son_10_donem.some(d => d.includes(41));
console.log("Is 41 in last 10 draws?", in_son_10_41);

// Look at how K2 is calculated normally
let f10 = son_10_donem.reduce((sum, d) => sum + (d.includes(41) ? 1 : 0), 0);
console.log("f10 for 41:", f10);
